src/v/cluster/topic_table.cc

```diff
@@ -719,7 +715,7 @@ topic_table::wait_for_changes(ss::abort_source& as) {
     }
 
     auto f = w->promise.get_future();
-    _waiters.push_back(std::move(w));
+    _waiter.swap(w);
```

Comment:
i think that `_waiter = std::move(w)` would be more idiomatic here.

Reply:
Not sure, I think your way may be more optimal though so I'll switch to it.